File: src/recommendation/risk_indicators/security_type_pack/security_type_ri.py
import os
import csv


def risk_value(isin):
  current_path = os.path.dirname(os.path.abspath(__file__))
  csv_path = os.path.join(current_path, 'all-securities.csv')
  with open(csv_path) as csv_file:
    secs = csv.reader(csv_file)
    sec = None
    for elem in secs:
      if elem[0] == isin:
        sec = elem
        break
    if sec is None:
      return None

    type = sec[1]

    switcher = {
      'STK': 4,
      'FND': 4,
      'IND': -1,
      'ETF': 4,
      'KNO': 5,
      'ZER': 4,
      'CUR': -1,
      'ETC': 1,
      'BND': 1,
      'CRYP': 5,
      'ETN': 5,
      'FUT': 5,
      'INT': -1,
      'OPT': 5,
      'RES': 3,
      'WNT': 5
    }

    risk_class = switcher.get(type)

    if risk_class is None or risk_class == -1:
      return None

    return (risk_class * 16.667) + (16.667 / 2)

File: src/recommendation/risk_indicators/security_type_pack/security-type-ri.js
const csv = require('csvtojson');
const path = require('path');

const myPath = path.join(__dirname, 'all-securities.csv');

// Risk percentage calculations are only based on the isin and should return a value between 0.0 and 100.0
module.exports = {
  riskPercentage : async (isin, callback) => {
    let sec = null;
    const securities = await csv().fromFile(myPath);
    for (let security of securities) {
      if (security.isin === isin) {
        sec = security;
        break;
      }
    }

    if (sec === null) return null;

    let riskClass = null;
    let type = sec.name;
    let unmatched = false;

    switch (type) {
      case 'STK':
        riskClass = 4;
        break;
      case 'FND':
        riskClass = 4;
        break;
      case 'IND':
        riskClass = -1;
        break;
      case 'ETF':
        riskClass = 4;
        break;
      case 'KNO':
        riskClass = 5;
        break;
      case 'ZER':
        riskClass = 4;
        break;
      case 'CUR':
        riskClass = -1;
        break;
      case 'ETC':
        riskClass = 1;
        break;
      case 'BND':
        riskClass = 1;
        break;
      case 'CRYP':
        riskClass = 5;
        break;
      case 'ETN':
        riskClass = 5;
        break;
      case 'FUT':
        riskClass = 5;
        break;
      case 'INT':
        riskClass = -1;
        break;
      case 'OPT':
        riskClass = 5;
        break;
      case 'RES':
        riskClass = 3;
        break;
      case 'WNT':
        riskClass = 5;
        break;
      default:
        console.log('Unmatched Security Type!');
        unmatched = true;
    }

    //return the value by invoking a callback
    //the 1st param. is error, just keep it null
    if (riskClass === -1 || unmatched === true) callback(null, null);
    else callback(null, (riskClass * 16.667 + 16.667 / 2));
  }
};

